File: javascript/leetcode/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    seen = {}
    for (var i = 0; i < nums.length; i++) {
        if (typeof seen[target - nums[i]] !== 'undefined') {
            return [i, seen[target - nums[i]]]
        }
        seen[nums[i]] = i
    }
    return false
};

File: python/leetcode/two-sum.py
class Solution:
    def twoSum(self, nums, target):
        seen = {}
        for i in range(len(nums)):
            if (target - nums[i]) in seen:
                res = []
                res.append(i)
                res.append(seen[target - nums[i]])
                return res
            seen[nums[i]] = i
        return False
